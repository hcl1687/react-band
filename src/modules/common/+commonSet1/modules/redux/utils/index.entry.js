import getStore from '../getStore'

const entry = () => {
  return {
    getStore
  }
}

export default {
  entry
}
