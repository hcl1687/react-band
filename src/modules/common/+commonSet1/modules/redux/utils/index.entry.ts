import getStore from '../getStore'

export default (): RB.IRBModule => {
  return {
    getStore
  }
}
