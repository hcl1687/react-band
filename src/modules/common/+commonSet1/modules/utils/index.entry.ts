function setStatic (key: string, value: string): Utils.IComponentDeco {
  return BaseComponent => {
    BaseComponent[key] = value
    return BaseComponent
  }
}

function setDisplayName (displayName: string): Utils.IComponentDeco {
  return setStatic('displayName', displayName)
}

function getDisplayName (Component: RB.IRBComponent|string): string|undefined {
  if (typeof Component === 'string') {
    return Component
  }

  if (!Component) {
    return undefined
  }

  return Component.displayName || Component.name || 'Component'
}

function wrapDisplayName (BaseComponent: RB.IRBComponent|string, hocName: string): string {
  return `${hocName}(${getDisplayName(BaseComponent)})`
}

function promisify (fun: (...args: Array<any>) => any): ((...args: Array<any>) => Promise<any>) {
  return (...args: Array<any>) => {
    return new Promise((resolve, reject) => {
      try {
        const ret = fun(...args)
        if (ret && ret.then) {
          ret.then(resolve).catch(reject)
        } else {
          resolve(ret)
        }
      } catch (e) {
        reject(e)
      }
    })
  }
}

function getUrlParameter (url: string, name: string): string {
  // eslint-disable-next-line
  name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]')
  const regex = new RegExp('[\\?&]' + name + '=([^&#]*)')
  const results = regex.exec(url)
  return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '))
}

const entry = (): RB.IRBModule => {
  return {
    setStatic,
    setDisplayName,
    getDisplayName,
    wrapDisplayName,
    promisify,
    getUrlParameter
  }
}

export default {
  entry
}
