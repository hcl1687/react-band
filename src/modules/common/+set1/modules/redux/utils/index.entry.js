import getStore from '../getStore'

export default () => {
  return {
    getStore
  }
}
