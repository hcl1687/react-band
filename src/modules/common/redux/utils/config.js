export default () => {
  return {
    name: 'reduxUtils'
  }
}
