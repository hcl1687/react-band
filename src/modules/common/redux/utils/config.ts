import { IRBLeafConfig } from '~/interface'

export default (): IRBLeafConfig => {
  return {
    name: 'reduxUtils'
  }
}
