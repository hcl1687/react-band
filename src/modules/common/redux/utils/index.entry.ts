import { IRBModule } from '~/interface'
import getStore from '../getStore'

export default (): IRBModule => {
  return {
    getStore
  }
}
