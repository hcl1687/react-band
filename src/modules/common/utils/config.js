export default (config) => {
  return {
    name: 'utils'
  }
}
