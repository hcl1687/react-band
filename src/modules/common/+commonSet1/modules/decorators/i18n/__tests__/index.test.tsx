import PropTypes, { InferProps } from 'prop-types'
import { mount, render } from 'enzyme'
import React from 'react'
import module from '../index.entry'
import utils from '~/../tests/utils/mockUtils'

const i18nDecoFactory = module.entry
const context: RB.IRBContext = {
  modules: {},
  i18ns: {},
  themes: {},
  packedModules: {},
  modulesConfig: {},
  routes: [],
  getModule: async (key) => {
    if (key === 'utils') {
      return utils
    }
  },
  options: {
    locale: 'en'
  }
}
function Test ({ __ }: InferProps<typeof Test.propTypes>) {
  return <div>{__('test')}</div>
}
Test.propTypes = {
  __: PropTypes.func.isRequired
}

describe('common/decorators/i18n', () => {
  it('should render correctly', async () => {
    const I18nDeco = await i18nDecoFactory(context)
    const i18n = {
      test: 'Test'
    }

    const TestWithI18nDeco = await I18nDeco({ i18n, name: 'test' }, { name: '@i18n' }, context)(Test)
    const wrapper = render(<TestWithI18nDeco />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should translate i18n correctly', async () => {
    const I18nDeco = await i18nDecoFactory(context)
    const i18n = {
      test: 'Test'
    }
    const TestWithI18nDeco = await I18nDeco({ i18n, name: 'test' }, { name: '@i18n' }, context)(Test)
    const wrapper = mount(
      <TestWithI18nDeco />
    )
    expect(wrapper.text()).toEqual('Test')
  })
})

describe('common/decorators/i18n/production', () => {
  const NODE_ENV = process.env.NODE_ENV
  beforeAll(() => {
    process.env.NODE_ENV = 'production'
  })
  afterAll(() => {
    process.env.NODE_ENV = NODE_ENV
  })
  it('should render correctly in production env', async () => {
    const I18nDeco = await i18nDecoFactory(context)
    const i18n = {
      test: 'Test'
    }
    const TestWithI18nDeco = await I18nDeco({ i18n, name: 'test' }, { name: '@i18n' }, context)(Test)
    const wrapper = render(<TestWithI18nDeco />)
    expect(wrapper).toMatchSnapshot()
  })
})
