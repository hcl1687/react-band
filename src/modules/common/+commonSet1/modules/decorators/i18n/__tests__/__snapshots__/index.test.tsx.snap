// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/decorators/i18n should render correctly 1`] = `
<div>
  Test
</div>
`;

exports[`common/decorators/i18n/production should render correctly in production env 1`] = `
<div>
  Test
</div>
`;
